moveit_kinematics/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/GreedyKCenters.h

```diff
@@ -95,7 +95,7 @@ class GreedyKCenters
     centers.push_back(std::uniform_int_distribution<size_t>{ 0, data.size() - 1 }(generator_));
     for (unsigned i = 1; i < k; ++i)
     {
-      unsigned ind;
+      unsigned ind = (unsigned)data.size();
```

Comment:
Is the range of ```data.size()``` narrower than that of ```unsinged```?
Is processing such as latching at the maximum value necessary?

Reply:
Since line 101 compares `unsigned j` and `data.size ()`, `data.size ()` is considered to be in the range of `unsigned`. (If `data.size()` exceeds `unsigned`, an infinite loop will occur at line 101)

Reply:
OK.